### PR TITLE
Added filtering

### PR DESCRIPTION
**Pull request by agnor99**
Also added base for ordering the outputs based on inputSettings

### CODE REVIEW

File: src/main/java/software/bernie/techarium/util/Both.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package software.bernie.techarium.util;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.Setter;
+
+@AllArgsConstructor
+@Getter
+@Setter
+public class Both<A,B> {
```
Comment (bernie-g) [inline src/main/java/software/bernie/techarium/util/Both.java:10]: https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/tuple/Pair.html

File: src/main/java/software/bernie/techarium/pipes/networks/ItemFilter.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package software.bernie.techarium.pipes.networks;
+
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.Items;
+
+public class ItemFilter extends Filter<ItemStack> {
+    private Item[] filtered = {Items.AIR,Items.AIR,Items.AIR,Items.AIR,Items.AIR};
```
Comment (bernie-g) [inline src/main/java/software/bernie/techarium/pipes/networks/ItemFilter.java:8]: I dunno if it's possible but it'd be nice to use `Ingredient` instead so we can use tags and other stuff

**Review by bernie-g — APPROVED** [review]
idk how in the fuck any of this shit works but im impressed

File: src/main/java/software/bernie/techarium/pipes/networks/ItemFilter.java (hunk 2 above)
Comment (agnor99) [reply src/main/java/software/bernie/techarium/pipes/networks/ItemFilter.java:8]: I'm not sure how you would insert the Tag into the filter in the gui? I thought putting items into fake slots would be easiest, but I can change it, if you come up with a way to display them

**Comment by agnor99** [comment]
> idk how in the fuck any of this shit works but im impressed

idk either /s

**Pull request merged**